### ISSUE TITLE
`dotnet add package` adds inappropriate metadata to Directory.Packages.props

### ISSUE THREAD

**Issue by AArnott** [NuGet/Home#13438]
### NuGet Product Used

dotnet.exe

### Product Version

8.0.204

### Impact

It's more difficult to complete my work

### Repro Steps & Context

I ran the following command line in the directory of an SDK style csproj where CPVM is turned on:

    dotnet add package MessagePackAnalyzer --version 3.0.54-alpha

This added the following xml to my csproj:
```xml
    <PackageReference Include="MessagePackAnalyzer">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
```

That looks good. But it also added the following xml to my Directory.Packages.props:
```xml
    <PackageVersion Include="MessagePackAnalyzer" Version="3.0.54-alpha">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageVersion>
```

PackageVersion items are not expected to have the metadata that the tool added, are they?

**Comment by jeffkl** [comment]
@AArnott you are correct, `dotnet add package` should not be adding that metadata to the `<PackageVersion />` item.  It looks like this would only happen for packages marked as a development dependency?  Does that sound right?

**Comment by AArnott** [comment]
Yes, that would explain how it hasn't been reported before.

**Comment by AArnott** [comment]
This is still broken.
